src/device.jl

```diff
@@ -18,6 +18,18 @@ function device_reset(dev::Integer)
     for p in todelete
         delete!(cuda_ptrs, p)
     end
+
+    pctx = get_pctx(dev)
+    # Cleanup CUDAdrv gc
+    if !CUDAdrv.can_finalize(pctx)
+        warn("Can't cleanly finalize pctx")
+        for obj in keys(filter((owner, ctx) -> ctx == pctx, CUDAdrv.finalizer_blocks))
+            CUDAdrv.finalize(Base.unsafe_pointer_to_objref(obj))
```

Comment:
This is a really ugly hack. I think we should stop encouraging `device_reset` and error in this case.